addons/account_peppol/wizard/service_wizard.py

```diff
@@ -0,0 +1,109 @@
+
+    wizard_id = fields.Many2one('account_peppol.service.wizard')
+    document_identifier = fields.Char()
+    document_name = fields.Char()
+    enabled = fields.Boolean(default=False)
+
+
+class PeppolServiceConfig(models.TransientModel):
+
+    _name = 'account_peppol.service.wizard'
+    _description = 'Peppol Services Wizard'
+
+    edi_user_id = fields.Many2one('account_edi_proxy_client.user', string='EDI user')
+    service_json = fields.Json(help="JSON representation of peppol services as retrieved from the peppol server.")
+    service_info = fields.Html(compute='_compute_service_info')
+    service_ids = fields.One2many(
+        comodel_name='account_peppol.service',
+        inverse_name='wizard_id',
+        readonly=False,
+    )
+
+    # -------------------------------------------------------------------------
+    # COMPUTES
+    # -------------------------------------------------------------------------
+
+    def _compute_service_info(self):
+
+        for wizard in self:
+            message = ''
+            supported_doctypes = self.env['res.company']._peppol_supported_document_types()
+            if (non_configurable := [
+                identifier for identifier in (wizard.service_json or {})
+                if identifier not in supported_doctypes
+            ]):
+                message = Markup('%s<ul>%s</ul>') % (_(
+                    "The following services are listed on your participant but cannot be configured here. "
+                    "If you wish to configure them differently, please contact support."),
+                    Markup().join(Markup('<li>%s</li>') % (
+                        wizard.service_json[identifier]['document_name'])
+                        for identifier in non_configurable
+                    ),
+                )
+            wizard.service_info = message
+
+    # -------------------------------------------------------------------------
+    # OVERRIDES
+    # -------------------------------------------------------------------------
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        """Get the selectable document types.
+
+        Synthesize a combination of locally available document types and those added to the user on
+        the IAP, add the relevant services.
+        """
+        wizards = super().create(vals_list)
+        supported_doctypes = self.env['res.company']._peppol_supported_document_types()
+        for wizard in wizards:
+            wizard.service_ids.create([{
+                'document_identifier': identifier,
+                'document_name': document_name,
+                'enabled': identifier in (wizard.service_json or {}),
+                'wizard_id': wizard.id,
+            } for identifier, document_name in supported_doctypes.items()])
+        return wizards
+
+    def write(self, vals):
+        """Interpret the write vals, and adapt them to make a request to update services."""
+        res = super().write(vals)
```

Comment:
I don't think its the right place to do that. Can't you call a specific method instead doing the service call?
Btw, the write is only called with the explicit values you provide. It will not work with computed field.
Also, you are making the service call at each write whatever the passed values.